app/ts/utils/ensUtils.ts

```diff
@@ -212,32 +212,32 @@ export const isPetalLockDeployed = async (account: AccountAddress | undefined) =
 	return deployedBytecode === expectedDeployedBytecode
 }
 
+export const deployPetalLockTransaction = () => {
+	const bytecode: `0x${ string }` = `0x${ petalLockContractArtifact.contracts['PetalLock.sol'].PetalLock.evm.bytecode.object }`
+	return { to: proxyDeployerAddress, data: bytecode } as const
+}
+
 export const deployPetalLock = async (account: AccountAddress) => {
 	if (await isPetalLockDeployed(account)) throw new Error('already deployed')
```

Comment:
At some point we may want to run these tests on a chain that already has PetalLock deployed, in which case we wouldn't want this to throw.  Would need more than just the below, like fixing the return type of this function.
```suggestion
export const ensurePetalLockDeployed = async (account: AccountAddress) => {
	if (await isPetalLockDeployed(account)) return true
```

Reply:
there's a separate function to test if the contracts are deployed (isPetalLockDeployed), and this is just intended to deploy it. If you want to have ensure contract, then that could be formed out of these too if thats needed

Reply:
Doesn't that make this call to `isPetalLockDeployed` redundant, since it should always be checked before this function is called?  The round trip to make that call isn't free, so doing it twice in sequence unnecessarily seems like an unnecessary slowdown.

Reply:
> Doesn't that make this call to `isPetalLockDeployed` redundant, since it should always be checked before this function is called? The round trip to make that call isn't free, so doing it twice in sequence unnecessarily seems like an unnecessary slowdown.

Its bit redundant yeah, maybe it can be removed, it's just a check if something has failed in checking it before

Reply:
I generally like redundancy, but in this case it is an "expensive" check as it requires one or more round trips to an RPC server.  I'm quite hesitant to have redundant checks that are expensive like that.  Especially since this check may be part of a simulation stack which makes it even more expensive.